pkg/scheduler/framework/plugins/podtopologyspread/common.go

```diff
@@ -135,10 +135,14 @@ func (pl *PodTopologySpread) filterTopologySpreadConstraints(constraints []v1.To
 
 func mergeLabelSetWithSelector(matchLabels labels.Set, s labels.Selector) labels.Selector {
 	mergedSelector := labels.SelectorFromSet(matchLabels)
-	if requirements, ok := s.Requirements(); ok {
-		for _, r := range requirements {
-			mergedSelector = mergedSelector.Add(r)
-		}
+
+	requirements, ok := s.Requirements()
```

Comment:
I guess this change doesn't matter anymore?

Reply:
Actually Yes. But the change is still needed to ensure that the logic is correct

Reply:
right, so this would ensure that empty label selector means match-none.

/lgtm
/approve